### PR TITLE
Add zones to map

### PR DESCRIPTION
**Pull request by jdudley1123**
## Context
Airtext will allow users to get notifications about air quality in certain zones. These zones are local authorities and other geographic regions. We want to display them on the map, with labels.

The boundaries of each zone are defined in geojson, and have been provided by CERC

Trello: https://trello.com/c/08sMFMOi/84-draw-zone-boundaries-on-map
https://trello.com/c/dKiwJnEI/106-spike-to-understand-optimal-map-layering-styling-and-transparency

## Changes in this PR
This PR adds the zones in full and simplified forms, and then adds the zones to map, showing boundaries and labels. The styling of the zones varies a little depending on whether they're top level (like 'Central London') or lower level (like 'Southwark').

To reduce the size of the geojson files, I've also added a script to simplify the files to an arbitrary level of accuracy.

### Before
image

### After
image
image

### CODE REVIEW

File: app/javascript/controllers/map_controller.js [hunk 1]
```diff
@@ -106,4 +110,76 @@ export default class MapController extends Controller {
...
+        style: function (feature) {
+          return {
+            color: "black", // stroke color
+            weight:
+              zoneBoundaryLayerLevelStyles[feature.properties.level].weight,
+            fill: false,
+          };
+        },
+        pane: "zones",
+      });
```
Comment (patrickjfl) [inline app/javascript/controllers/map_controller.js:139]: am I reading this right that we're adding a new layer to the map for each zone? If so, is that necessary / is there some way to incorporate the boundaries of both levels of geography into two layers? Or am I showing my ignorance of how the map works?

**Review by patrickjfl — APPROVED** [review]
A bit confused about the layers, but overall looks good!

Also - how are we centring the labels? Is it based on some calculation of the average of all the points? Just because some of them look a little off (e.g. Walthamstow).

File: app/javascript/controllers/map_controller.js (hunk 1 above)
Comment (jdudley1123) [reply app/javascript/controllers/map_controller.js:139]: Yes that's right, each one is a separate layer (within the `zones` pane div). I think in theory you could combine them all into one, but I don't think there is any drawback of having them all as separate layers.

**Comment by jdudley1123** [comment]
> A bit confused about the layers, but overall looks good!
> 
> Also - how are we centring the labels? Is it based on some calculation of the average of all the points? Just because some of them look a little off (e.g. Walthamstow).

Yeah the centring a bit janky. It takes a bounding box of the zone and then works out the centre (not sure of the exact calculation). For zones with sticky out bits that will not be the visual centre. I think there's also an issue with that then being the top left of the label rather than the centre.

I think it's good enough for now, but I'd like to revisit it if we have time later in the project to add some polish.

https://trello.com/c/3mzOYVqs/114-improve-positioning-of-zone-labels

**Pull request merged**